llama_stack/apis/common/content_types.py

```diff
@@ -72,10 +72,8 @@ class TextContentItem(BaseModel):
 )
 
 # accept a single "str" as a special case since it is common
-InterleavedContent = register_schema(
-    Union[str, InterleavedContentItem, List[InterleavedContentItem]],
-    name="InterleavedContent",
-)
+InterleavedContentType = Union[str, InterleavedContentItem, list[InterleavedContentItem]]
```

Comment:
Hm, we use the `Annotated[Union[...]` and `register_schema` everywhere in the codebase. We need to find a cleaner solution here because these changes make everything less ergonomic and bloat-y.

Reply:
Yeah I figured already. :) I think I have an idea how this should be handled. Let me take this for a test. I will make it a draft and update later.